kernel/common/string.rb

```diff
@@ -42,8 +42,9 @@ def clone
     alias_method :dup, :clone
   end
 
-  def initialize(arg = undefined)
+  def initialize(arg = undefined, encoding: nil)
     replace arg unless undefined.equal?(arg)
+    self.force_encoding(encoding) if encoding
```

Comment:
There is no reason for that I was not aware of the `encode!` method.
Should I change it to use `encode!` ?